### PR TITLE
Docs: add data passing in components doc

### PR DESCRIPTION
**Pull request by FogDong**
Hold this until this [PR](https://github.com/oam-dev/kubevela/pull/2236) is merged.

### CODE REVIEW

File: docs/end-user/component-dependency-parameter.md [hunk 1]
```diff
@@ -2,4 +2,67 @@
 title:  Data Pass Between Components 
 ---
 
-WIP
\ No newline at end of file
+This section will introduce how to pass data between components.
+
+## How to use
+
+If we want to apply a WordPress server with data stored in a MySQL database, we need to pass the MySQL address to WordPress.
```
Comment (hongchaodeng) [inline docs/end-user/component-dependency-parameter.md]:
```suggestion
In the following we will apply a WordPress server with the MySQL address passed from a MySQL component.
```

**Review by hongchaodeng — APPROVED** [review]
lgtm after nit

File: docs/end-user/component-dependency-parameter.md [hunk 2]
```diff
@@ -2,4 +2,67 @@
...
+  name: wordpress-with-mysql
+  namespace: default
+spec:
+  components:
+    - name: mysql
+      type: helm
+      outputs:
+        # the output is the mysql service address
+        - name: mysql-svc
+          exportKey: output.metadata.name + ".default.svc.cluster.local"
```
Comment (hongchaodeng) [inline docs/end-user/component-dependency-parameter.md:24]: Isn't this not a key but a value?

Comment (wonderflow) [reply docs/end-user/component-dependency-parameter.md:24]: I think this demo not work, please confirm @FogDong

Comment (hongchaodeng) [reply docs/end-user/component-dependency-parameter.md:24]: Talked to @FogDong . She will rectify it.

Comment (FogDong) [reply docs/end-user/component-dependency-parameter.md:24]: The exportKey can be made up of key and value, the test has been passed locally.

Ref: https://github.com/oam-dev/kubevela/blob/master/pkg/controller/core.oam.dev/v1alpha2/application/generator_test.go#L103

**Review by wonderflow — APPROVED** [review]

**Pull request merged**